crates/uv-auth/src/middleware.rs

```diff
@@ -48,6 +49,54 @@ impl NetrcMode {
     }
 }
 
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq)]
+pub enum AuthPolicy {
+    /// Attempt to authenticate if needed.
+    #[default]
+    Auto,
+    /// Always attempt to authenticate.
+    Always,
+    /// Never attempt to authenticate.
+    Never,
+}
+
+#[derive(Debug, Default, Clone, Eq, PartialEq)]
+pub struct UrlAuthPolicies(FxHashMap<Url, AuthPolicy>);
+
+impl UrlAuthPolicies {
```

Comment:
Nit: I'd put this and related in a `uv-auth::policy` module instead of adding it to the `middleware` module (mostly to match the existing level of granularity). Kind of weird the NetrcMode is in here already.